tests/functional/event/test_webhook.py

```diff
@@ -0,0 +1,114 @@
+# -*- coding: utf-8 -*-
+
+# Copyright (C) 2018 OpenIO SAS, as part of OpenIO SDS
+#
+# This library is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 3.0 of the License, or (at your option) any later version.
+#
+# This library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this library.
+
+
+import requests
+import time
+
+from oio.api.object_storage import ObjectStorageApi
+from tests.utils import BaseTestCase
+
+
+WEB = None
+
+
+class TestMeta2EventsEmission(BaseTestCase):
+    def setUp(self):
+        super(TestMeta2EventsEmission, self).setUp()
+
+        self.acct = 'AccountWebhook%f' % time.time()
+        self.cnt_name = 'TestWebhookEvents%f' % time.time()
+        self.obj_name = 'obj%f' % time.time()
+        self.storage_api = ObjectStorageApi(self.ns)
+        self._clean()
+
+    def _get(self, success=True, timeout=2, event_id=None):
+        path = '%s/%s/%s' % (self.acct, self.cnt_name, self.obj_name)
+        start = time.time()
+        while time.time() - start < timeout:
+            res = requests.get('http://127.0.0.1:9081/' + path)
+            if success and res.status_code == requests.codes.ok:
+                if not event_id or event_id < res.json()['eventId']:
+                    return res, res.json()
+            if not success and res.status_code == requests.codes.not_found:
+                return res, None
+
+            time.sleep(0.1)
+        # fixme
+        assert("Timeout waiting webhook event")
+
+    def _clean(self):
+        if self.conf.get('webhook', ''):
+            requests.post('http://127.0.0.1:9081/PURGE')
+
+    def _add(self, data, properties=None):
+        self.storage_api.object_create(self.acct, self.cnt_name,
+                                       data=data, obj_name=self.obj_name,
+                                       properties=properties)
+        ret, data = self._get()
+        self.assertEqual(ret.status_code, requests.codes.ok)
+        return ret, data
+
+    def _remove(self):
+        self.storage_api.object_delete(self.acct, self.cnt_name, self.obj_name)
+        ret, data = self._get(success=False)
+        self.assertEqual(ret.status_code, requests.codes.not_found)
+        return ret, data
+
+    def test_content_add(self):
+        if not self.conf.get('webhook', ''):
+            self.skipTest('webhook is required')
```

Comment:
Minor: since all these tests require webhook, these 2 lines can be put in `setUp()`.